Dịch "lenet_vn" - Phần 4 (#1306)
aivivn/d2l-vn: chapter_convolutional-neural-networks/lenet_vn.md

Then, the fully-connected layer reduces dimensionality layer by layer, until emitting an output that matches the number of image classes.
-->

> Xin hãy chú ý rằng, chiều cao và chiều rộng của đặc trưng thu được từ việc học biểu diễn tại mỗi tầng trong toàn bộ khối tính chập sẽ bị giảm đi (so với chiều cao và chiều rộng của đặc trưng ở tầng trước). Tầng tính chập đầu tiên sử dụng một kernel với chiều cao và chiều rộng là $5$, và tiếp theo đó là đệm thêm vào $2$ đơn vị điểm ảnh để bù trừ cho sự giảm đi kích thước của đặc trưng đầu ra so với kích thước ban đầu của nó. Trong khi đó tầng tính chập thứ hai cũng dùng cùng một kernel với kích thước là $5 x 5$ mà không có sử dụng giá trị đệm thêm vào, dẫn đến việc giảm cả chiều cao và chiều rộng bằng một giá trị là 4 đơn vị điểm ảnh. Ngoài ra, mỗi tầng gộp sẽ làm giảm đi một nửa chiều cao và chiều rộng của đặc trưng ánh xạ đầu vào. Tuy nhiên, khi chúng ta đi từ thứ tự từ dưới lên trên của các tầng chồng lên nhau, số kênh sẽ tăng lần lượt theo từng tầng, từ 1 kênh của dữ liệu đầu vào cho lên tới 6 kênh sau khi đi qua tầng tính chập thứ nhất và 16 kênh sau khi đi qua tầng tính chập thứ hai. Sau đó, tầng kết nối đầy đủ làm giảm số chiều lần lượt qua từng tầng cho đến khi tạo thành một đầu ra khớp với số lượng các loại nhãn của hình ảnh. 

Reviewer: ```suggestion
Xin hãy chú ý rằng, chiều cao và chiều rộng của đặc trưng thu được từ việc học biểu diễn tại mỗi tầng trong toàn bộ khối tính chập sẽ bị giảm đi (so với chiều cao và chiều rộng của đặc trưng ở tầng trước). 
Tầng tính chập đầu tiên sử dụng một kernel với chiều cao và chiều rộng là $5$, và tiếp theo đó là đệm thêm vào $2$ đơn vị điểm ảnh để bù trừ cho sự giảm đi kích thước của đặc trưng đầu ra so với kích thước ban đầu của nó. 
Trong khi đó tầng tính chập thứ hai cũng dùng cùng một kernel với kích thước là $5 x 5$ mà không có sử dụng giá trị đệm thêm vào, dẫn đến việc giảm cả chiều cao và chiều rộng bằng một giá trị là 4 đơn vị điểm ảnh. 
Ngoài ra, mỗi tầng gộp sẽ làm giảm đi một nửa chiều cao và chiều rộng của đặc trưng ánh xạ đầu vào. 
Tuy nhiên, khi chúng ta đi từ thứ tự từ dưới lên trên của các tầng chồng lên nhau, số kênh sẽ tăng lần lượt theo từng tầng, từ 1 kênh của dữ liệu đầu vào cho lên tới 6 kênh sau khi đi qua tầng tính chập thứ nhất và 16 kênh sau khi đi qua tầng tính chập thứ hai. 
Sau đó, tầng kết nối đầy đủ làm giảm số chiều lần lượt qua từng tầng cho đến khi tạo thành một đầu ra khớp với số lượng các loại nhãn của hình ảnh. 
```

Mình tách dòng ra

Then, the fully-connected layer reduces dimensionality layer by layer, until emitting an output that matches the number of image classes.
-->

> Xin hãy chú ý rằng, chiều cao và chiều rộng của đặc trưng thu được từ việc học biểu diễn tại mỗi tầng trong toàn bộ khối tính chập sẽ bị giảm đi (so với chiều cao và chiều rộng của đặc trưng ở tầng trước). 

Reviewer 2: nhiều người dùng luôn `biểu diễn` là danh từ rồi, mình thấy nên như này cho gọn
```suggestion
Xin hãy chú ý rằng, chiều cao và chiều rộng của biểu diễn tại mỗi tầng trong toàn bộ khối tích chập sẽ bị giảm đi (so với chiều cao và chiều rộng của biểu diễn ở tầng trước). 
```

> Tầng tích chập đầu tiên sử dụng một kernel với chiều cao và chiều rộng là $5$ rồi đệm thêm $2$ đơn vị điểm ảnh để bù trừ cho sự giảm đi kích thước của đặc trưng đầu ra so với kích thước ban đầu của nó. 

Reviewer 3: Update lại phần kernel này sau khi PR này (#1363) được thống nhất nhé @DacDinh147

Author: @duythanhvn em đã update rồi anh.

> Tầng tính chập đầu tiên sử dụng một kernel với chiều cao và chiều rộng là $5$, và tiếp theo đó là đệm thêm vào $2$ đơn vị điểm ảnh để bù trừ cho sự giảm đi kích thước của đặc trưng đầu ra so với kích thước ban đầu của nó. 

Reviewer 2: ```suggestion
Tầng tích chập đầu tiên sử dụng một kernel với chiều cao và chiều rộng là $5$ rồi đệm thêm $2$ đơn vị điểm ảnh để bù trừ cho sự giảm đi kích thước của đặc trưng đầu ra so với kích thước ban đầu của nó. 
```

> Trong khi đó tầng tính chập thứ hai cũng dùng cùng một kernel với kích thước là $5 x 5$ mà không có sử dụng giá trị đệm thêm vào, dẫn đến việc giảm cả chiều cao và chiều rộng bằng một giá trị là 4 đơn vị điểm ảnh. 

Reviewer 2: ```suggestion
Trong khi đó tầng tích chập thứ hai cũng dùng cùng một kernel với kích thước là $5 x 5$ mà không có sử dụng giá trị đệm thêm vào, dẫn đến việc chiều cao và chiều rộng giảm đi 4 đơn vị điểm ảnh. 
```

Ngoài ra, mỗi tầng gộp sẽ làm giảm đi một nửa chiều cao và chiều rộng của đặc trưng ánh xạ đầu vào. 

> Tuy nhiên, khi chúng ta đi từ thứ tự từ dưới lên trên của các tầng chồng lên nhau, số kênh sẽ tăng lần lượt theo từng tầng, từ 1 kênh của dữ liệu đầu vào cho lên tới 6 kênh sau khi đi qua tầng tính chập thứ nhất và 16 kênh sau khi đi qua tầng tính chập thứ hai. 

Reviewer 2: ```suggestion
Tuy nhiên, khi chúng ta đi từ thứ tự từ dưới lên trên của các tầng chồng lên nhau, số kênh sẽ tăng lần lượt theo từng tầng, từ 1 kênh của dữ liệu đầu vào lên tới 6 kênh sau khi đi qua tầng tích chập thứ nhất và 16 kênh sau khi đi qua tầng tích chập thứ hai. 
```

> Sau đó, tầng kết nối đầy đủ làm giảm số chiều lần lượt qua từng tầng cho đến khi tạo thành một đầu ra khớp với số lượng các loại nhãn của hình ảnh. 

Reviewer 2: ```suggestion
Sau đó, tầng kết nối đầy đủ làm giảm số chiều lần lượt qua từng tầng cho đến khi tạo thành một đầu ra khớp với số lượng lớp của hình ảnh. 
```

## Data Acquisition and Training
-->

## Thu thập và huấn luyện dữ liệu 

<!--
Now that we have implemented the model, we might as well run some experiments to see what we can accomplish with the LeNet model.
We will use Fashion-MNIST as our dataset.
It is more challenging than the original MNIST dataset while it has the same shape ($28\times28$ images).
-->

> Sau khi triển khai và xây dựng xong mô hình, chúng ta thực hiện một số thử nghiệm để xem chúng ta có thể đạt được kết quả gì với mô hình LeNet. Chúng ta sẽ sử dụng Fashion-MNIST là bộ dữ liệu huấn luyện. Bộ dữ liệu này sẽ thử thách hơn so với bộ MNIST trong bài toán phân loại mặc dù chúng có cùng kích thước của hình ảnh huấn luyện là $28\times28$. 

Reviewer: ```suggestion
Sau khi triển khai và xây dựng xong mô hình, chúng ta thực hiện một số thử nghiệm để xem chúng ta có thể đạt được kết quả gì với mô hình LeNet. Chúng ta sẽ sử dụng Fashion-MNIST là bộ dữ liệu huấn luyện. 
Bộ dữ liệu này sẽ thử thách hơn so với bộ MNIST trong bài toán phân loại mặc dù chúng có cùng kích thước của hình ảnh huấn luyện là $28\times28$. 
```

```{.python .input}
batch_size = 256
train_iter, test_iter = d2l.load_data_fashion_mnist(batch_size=batch_size)
```

<!--
While convolutional networks may have few parameters, they can still be significantly more expensive to compute than a similarly deep multilayer perceptron so if you have access to a GPU, this might be a good time to put it into action to speed up training.
-->

> Trong khi những mạng tính chập có thể có số lượng tham số không lớn, nhưng chúng vẫn cần tiêu tốn tài nguyên tính toán hơn nhiều so với việc học sâu với nhiều tầng perceptron, vì vậy, nếu bạn có thể sử dụng được GPU, đây là thời điểm thích hợp để dùng nó vào việc tăng tốc thời gian huấn luyện. 

Reviewer: ```suggestion
Trong khi những mạng tính chập có thể có số lượng tham số không lớn, nhưng chúng vẫn cần tiêu tốn tài nguyên tính toán hơn nhiều so với perceptron sâu đa tầng, vì vậy, nếu bạn có thể sử dụng được GPU, đây là thời điểm thích hợp để dùng nó vào việc tăng tốc thời gian huấn luyện. 
```

## Data Acquisition and Training
-->

> ## Thu thập và huấn luyện dữ liệu 

Reviewer 2: ```suggestion
## Thu thập dữ liệu và Huấn luyện
```

<!--
Now that we have implemented the model, we might as well run some experiments to see what we can accomplish with the LeNet model.
We will use Fashion-MNIST as our dataset.
It is more challenging than the original MNIST dataset while it has the same shape ($28\times28$ images).
-->

> Sau khi triển khai và xây dựng xong mô hình, chúng ta thực hiện một số thử nghiệm để xem chúng ta có thể đạt được kết quả gì với mô hình LeNet. Chúng ta sẽ sử dụng Fashion-MNIST là bộ dữ liệu huấn luyện. 

Reviewer 2: mọi ng quy ước dùng từ `triển khai` cho từ `deploy`
```suggestion
Sau khi xây dựng xong mô hình, chúng ta thực hiện một số thử nghiệm để xem chúng ta có thể đạt được kết quả gì với mô hình LeNet. 
Chúng ta sẽ sử dụng tập dữ liệu Fashion-MNIST.
```

> Bộ dữ liệu này sẽ thử thách hơn so với bộ MNIST trong bài toán phân loại mặc dù chúng có cùng kích thước của hình ảnh huấn luyện là $28\times28$. 

Reviewer 2: ```suggestion
Việc phân loại tập dữ liệu này sẽ khó hơn so với tập MNIST gốc mặc dù chúng cũng chứa các ảnh có cùng kích thước là $28\times28$. 
```

```{.python .input}
batch_size = 256
train_iter, test_iter = d2l.load_data_fashion_mnist(batch_size=batch_size)
```

<!--
While convolutional networks may have few parameters, they can still be significantly more expensive to compute than a similarly deep multilayer perceptron so if you have access to a GPU, this might be a good time to put it into action to speed up training.
-->

> Trong khi những mạng tính chập có thể có số lượng tham số không lớn, nhưng chúng vẫn cần tiêu tốn tài nguyên tính toán hơn nhiều so với perceptron sâu đa tầng, vì vậy, nếu bạn có thể sử dụng được GPU, đây là thời điểm thích hợp để dùng nó vào việc tăng tốc thời gian huấn luyện. 

Reviewer 2: ```suggestion
Dù mạng tích chập có thể có số lượng tham số không lớn, chúng vẫn tiêu tốn tài nguyên tính toán hơn nhiều so với perceptron sâu đa tầng. 
Vì vậy, nếu bạn có GPU thì đây là thời điểm thích hợp để dùng nó vào việc tăng tốc quá trình huấn luyện. 
```

Reviewer 4: ```suggestion
Dù mạng tích chập có thể có số lượng tham số không lớn, chúng vẫn tiêu tốn tài nguyên tính toán hơn nhiều so với perceptron sâu đa tầng. 
Vì vậy, nếu bạn có GPU thì đây là thời điểm thích hợp để dùng nó tăng tốc quá trình huấn luyện. 
```
sửa của @minhduc0711 chút.

Reviewer 2: như thế lại thấy thiếu thiếu, như này được k nhỉ
```suggestion
Dù mạng tích chập có thể có số lượng tham số không lớn, chúng vẫn tiêu tốn tài nguyên tính toán hơn nhiều so với perceptron sâu đa tầng. 
Vì vậy, nếu bạn có GPU thì đã đến lúc dùng nó để tăng tốc quá trình huấn luyện. 
```

<!--
For evaluation, we need to make a slight modification to the `evaluate_accuracy` function that we described in :numref:`sec_softmax_scratch`.
Since the full dataset lives on the CPU, we need to copy it to the GPU before we can compute our models.
This is accomplished via the `as_in_ctx` function described in :numref:`sec_use_gpu`.
-->

> Để thực hiện việc đánh giá hiệu quả của mô hình huấn luyện, chúng ta cần làm một điều chỉnh nhỏ cho hàm `evaluate_accuracy` mà chúng ta đã mô tả ở phần :numref:`sec_softmax_scratch`.

Reviewer 2: ```suggestion
Để đánh giá mô hình, chúng ta cần phải điều chỉnh hàm `evaluate_accuracy` mà chúng ta đã mô tả ở phần :numref:`sec_softmax_scratch` một chút.
```

Bởi vì toàn bộ tập dữ liệu đang nằm trên CPU, chúng ta cần sao chép nó lên GPU trước khi chúng ta có thể thực hiện tính toán với mô hình.

> Việc này được hoàn thành thông qua việc dùng hàm `as_in_ctx` đã được mô tả ở phần :numref:`sec_use_gpu`.  

Reviewer 2: ```suggestion
Việc này được thực hiện thông qua việc gọi hàm `as_in_ctx` mô tả ở phần :numref:`sec_use_gpu`.  
```
